programs/Makefile

```diff
@@ -77,6 +79,10 @@ ifdef PTHREAD
 APPS +=	ssl/ssl_pthread_server$(EXEXT)
 endif
 
+ifdef TEST_CPP
+APPS += test/cpp_dummy_build$(EXEXT)
```

Comment:
If you run `make TEST_CPP=1` and later `make clean`, then `test/cpp_dummy_build` doesn't get cleaned. There's a similar preexisting defect with `ssl/ssl_pthread_server`. The defect with `test/cpp_dummy_build` is worse because we include it in our test scripts.

I won't block the PR for that, but please, in order of preference, either:

* arrange for `make clean` to remove `test/cpp_dummy_build` (and `ssl/ssl_pthread_server`) regardless of the value of `TEST_CPP`;
* or file an issue.

Reply:
@AndrzejKurek is out for a couple of weeks, so I've raised this as issue #1862, so we can at least merge what we have, and fix this later, as it's a minor issue,